### PR TITLE
add .gitlab-ci.yml to build vpp image in gitlab

### PR DESCRIPTION
**Pull request by dansl19**
Add .gitlab-ci.yml in order to use CI/CD pipeline in GitLab.
Here we have two parts:
Github repo: Code management.
GitLab repo: CI/CD pipeline for code which is managed on Github.
Plan is continue to use Github for code management and GitLab for CI/CD pipeline.
GitLab repo : https://gitlab.com/travelping/vpp-github

You can find the pipeline in GitLab here : https://gitlab.com/travelping/vpp-github/pipelines

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+stages:
+  - build
+
+variables:
+  CONTAINER_IMAGE: ergw/vpp-test
```
Comment (RoadRunnr) [inline .gitlab-ci.yml]: why is this called vpp-test ?

Comment (dansl19) [reply .gitlab-ci.yml]: only for the test purpose , if there is no objection I will change the name to "vpp"

**Comment by tlnd** [comment]
@dansl19 Can you remove the failing container build from `.travis-ci.yaml` so that no confusion remains regarding CI?
And please also update the README.md to provide a note about the fact that CI is at gitlab.com providing a respective link.

**Comment by tlnd** [comment]
...and maybe also a direct link to the container image registry.

**Comment by dansl19** [comment]
@tlnd  I have commented build and upload steps in travis.yaml file (I couldn't find .travis-ci.yaml) and added description to the README file for build and upload process of VPP image with GitLab CI/CD pipeline.

**Comment by tlnd** [comment]
@dansl19 What I mean is, that with this MR, you should remove the CI from `.travis.yaml`, as it is introduced in gitlab.com. The only change I see in this MR is in `.gitlab-ci.yaml`. Hence, I also don't see the changes in the README you mention.

**Comment by dansl19** [comment]
prepared everything but didn't pushed my changes. @tlnd  do we need to keep .travis.yml file there or it can be deleted?

File: README.md [hunk 2]
```diff
@@ -151,3 +151,25 @@ advanced building strategies and other development notes.
...
+
+1. GitHub :  https://github.com/travelping/vpp    # This repository is used for version control of VPP code.
+2. GitLab :  https://gitlab.com/travelping/vpp-github  #  This repository is configured as mirror repository with upstream to GitHub repository. This part is used for CI/CD pipeline only.
+3. DockerHubRegistry: https://hub.docker.com/u/ergw/dashboard/    # After successful build new VPP Docker Image will be uploaded to this registry. 
+
+
+Build and Upload Pipeline. 
+
+1. Commit on https://github.com/travelping/vpp repository will be mirrored to the https://gitlab.com/travelping/vpp-github and triger CI/CD PipeLine on GitLab. Mirroring (Sync) between GitHub and GitLab occurs automatically every 30 minutes or can be trioggered manually. 
+2. GitLab CI/CD will trigger build of Docker Image, build proccess will be executed on GtiLab-Runner (Agent). GtiLab-Runner is running on "hamster" K8S cluster. Use the following URL https://gitlab.com/groups/travelping/-/settings/ci_cd to see available GitLab-Runner.
```
Comment (tlnd) [inline README.md]: Don't expose "hamster". You can state that it is build on dedicated runner instances. You also don't need to refer to those settings. This is documentation at github.com, hence for public and not just for internal.

File: README.md [hunk 3]
```diff
@@ -151,3 +151,25 @@ advanced building strategies and other development notes.
...
+
+
+
+### Build and Upload Docker VPP Image 
+
+Build and Upload VPP Docker Image comprise of three parts: 
+
+1. GitHub :  https://github.com/travelping/vpp    # This repository is used for version control of VPP code.
+2. GitLab :  https://gitlab.com/travelping/vpp-github  #  This repository is configured as mirror repository with upstream to GitHub repository. This part is used for CI/CD pipeline only.
+3. DockerHubRegistry: https://hub.docker.com/u/ergw/dashboard/    # After successful build new VPP Docker Image will be uploaded to this registry.
```
Comment (tlnd) [inline README.md]: Directly link to the respective image.

File: .travis.yml [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
-  - docker build -t $BUILD_IMAGE -f extras/docker/Dockerfile .
-
-after_success:
-  - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
-  - export TAG=`if [ "$TRAVIS_EVENT_TYPE" == "pull_request" ]; then echo PR-$TRAVIS_PULL_REQUEST_BRANCH ; else echo $TRAVIS_BRANCH ; fi`
-  - if [ "$TAG" == "feature/gtp-up" ]; then export TAG="latest"; fi
-  - docker tag $BUILD_IMAGE $BUILD_IMAGE:$TAG
-  - docker push $BUILD_IMAGE:$TAG
-  - echo "docker push done"
+#language: c
```
Comment (tlnd) [inline .travis.yml]: Don't comment, but remove.

**Review by tlnd — COMMENTED** [review]
We should remove the file completely, if nothing valuable ramins and @RoadRunnr don't complains.

**Comment by dansl19** [comment]
travis.ci file has been removed and I adapted description in README file accordingly to the public repository.

**Comment by RoadRunnr** [comment]
please, squash the commits into a single commit.

**Comment by dansl19** [comment]
squashed  into single commit

**Pull request merged**